### PR TITLE
feat: [CDS-85162]: FE changes for AWS connector OIDC token support

### PR DESCRIPTION
**Pull request by shivanand-harness**
### Summary

Jira Ticket: https://harness.atlassian.net/browse/CDS-85162

#### Screenshots

https://github.com/harness/harness-core-ui/assets/148759765/fb798e7d-70cc-42c5-85b3-3ff9cbc16cf0

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by shivanand-harness** [comment]
retrigger cypress-cd

**Comment by shivanand-harness** [comment]
retrigger cypress-rest

**Review by PratyushGar — APPROVED** [review]

File: src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/StepAuth/StepAWSAuthentication.tsx [hunk 1]
```diff
@@ -56,12 +57,48 @@ const StepAWSAuthentication: React.FC<StepProps<StepAWSAuthenticationProps> & Co
...
+  const getShouldShowCrossAccountAccess = (delegateType: CredentialType['key']) => {
+    switch (delegateType) {
+      case DelegateTypes.DELEGATE_OIDC:
+        return false
+      default:
+        return true
+    }
+  }
+
+  const radioGroupItems = useMemo(() => {
```
Comment (avinashmadhwani02) [inline src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/StepAuth/StepAWSAuthentication.tsx:77]: we can avoid useMemo here? there are no heavy calculations. 
useMemo itself does the deep comparison

File: src/modules/27-platform/connectors/pages/connectors/utils/ConnectorListViewUtils.tsx [hunk 2]
```diff
@@ -32,17 +32,31 @@ const textRenderer = (value: string): JSX.Element => {
   )
 }
 
+const stringRenderer = (titleStringId: StringKeys): JSX.Element | string => {
+  return (
+    <div className={classNames(css.name, css.flex)}>
```
Comment (avinashmadhwani02) [inline src/modules/27-platform/connectors/pages/connectors/utils/ConnectorListViewUtils.tsx:37]: do we need this div if titleStringId is not defined?

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/StepAuth/StepAWSAuthentication.tsx [hunk 3]
```diff
@@ -56,12 +57,48 @@ const StepAWSAuthentication: React.FC<StepProps<StepAWSAuthenticationProps> & Co
...
+  const { CDS_AWS_OIDC_AUTHENTICATION } = useFeatureFlags()
   const [initialValues, setInitialValues] = useState(defaultInitialFormData)
   const [loadingConnectorSecrets, setLoadingConnectorSecrets] = useState(props.isEditMode)
   const hideHeaderAndNavBtns = shouldHideHeaderAndNavBtns(context)
   useConnectorWizard({
     helpPanel: props.helpPanelReferenceId ? { referenceId: props.helpPanelReferenceId, contentWidth: 900 } : undefined
   })
+
+  const getShouldShowCrossAccountAccess = (delegateType: CredentialType['key']) => {
+    switch (delegateType) {
```
[the remark this answers is not in the document]
Comment (shivanand-harness) [reply src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/StepAuth/StepAWSAuthentication.tsx:69]: not required as off now... but if want to hide this section for multiple delegateTypes then switch will be more readable so added.

File: src/modules/27-platform/connectors/pages/connectors/utils/ConnectorListViewUtils.tsx (hunk 2 above)
Comment (shivanand-harness) [reply src/modules/27-platform/connectors/pages/connectors/utils/ConnectorListViewUtils.tsx:38]: resolved

File: src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/StepAuth/StepAWSAuthentication.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (shivanand-harness) [reply src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/StepAuth/StepAWSAuthentication.tsx]: removed

File: src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx [hunk 4]
```diff
@@ -324,4 +332,184 @@ describe('Create AWS connector Wizard', () => {
...
+              type: 'OidcAuthentication'
+            },
+            executeOnDelegate: false,
+            proxy: undefined
+          },
+          tags: {},
+          type: 'Aws'
+        }
+      },
+      expect.anything()
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx:512]: I'm not a big fan of `expect.anything()` as it's very broad (read: superset of everything) and if something changes the next time, this won't catch it.

Comment (shivanand-harness) [reply src/modules/27-platform/connectors/components/CreateConnector/AWSConnector/__test__/CreateAWSConnector.test.tsx:512]: Agree, but in this case we only care about the body sending to API and 2nd param is query param which we can ignore

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**